core/capabilities/ccip/ccipevm/commitcodecv2.go

```diff
@@ -0,0 +1,109 @@
+package ccipevm
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+	commoncodec "github.com/smartcontractkit/chainlink-common/pkg/codec"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/codec"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+)
+
+const commitReportABI = `[{"components":[{"name":"chainSel","type":"uint64","internalType":"uint64"},{"name":"onRampAddress","type":"bytes","internalType":"bytes"},{"name":"seqNumsRange","type":"uint64[2]","internalType":"uint64[2]"},{"name":"merkleRoot","type":"bytes32","internalType":"bytes32"}],"name":"merkleRoots","type":"tuple[]","internalType":"struct MerkleRootChain[]"},{"components":[{"components":[{"name":"tokenID","type":"string","internalType":"string"},{"name":"price","type":"uint256","internalType":"uint256"}],"name":"tokenPriceUpdates","type":"tuple[]","internalType":"struct TokenPrice[]"},{"components":[{"name":"chainSel","type":"uint64","internalType":"uint64"},{"name":"gasPrice","type":"uint256","internalType":"uint256"}],"name":"gasPriceUpdates","type":"tuple[]","internalType":"struct GasPriceChain[]"}],"name":"priceUpdates","type":"tuple","internalType":"struct PriceUpdates"},{"components":[{"name":"r","type":"bytes32","internalType":"bytes32"},{"name":"s","type":"bytes32","internalType":"bytes32"}],"name":"rmnSignatures","type":"tuple[]"}]`
```

Comment:
Would be generated from the contracts eventually?

Reply:
Yes, that would be ideal. The e existing generated [ABI](https://github.com/smartcontractkit/chainlink/blob/86d5f9993ef208ef0a310a781b198563f4ae7d55/core/gethwrappers/ccip/generated/report_codec/report_codec.go#L101) seems match with the format from generated [MerkleRoot](https://github.com/smartcontractkit/chainlink/blob/86d5f9993ef208ef0a310a781b198563f4ae7d55/core/gethwrappers/ccip/generated/ccip_encoding_utils/ccip_encoding_utils.go#L61), not with the original report [format](https://github.com/smartcontractkit/chainlink-ccip/blob/8b4b62f86e5716b17253d7529e0e688496deefdc/pkg/types/ccipocr3/plugin_commit_types.go#L43).